lib/components/base/Pagination.vue

```diff
@@ -50,65 +50,60 @@
   </div>
 </template>
 <script setup>
+import { computed } from 'vue'
 import { GapIcon, LeftArrowIcon, RightArrowIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  props: {
-    page: {
-      type: Number,
-      default: 1,
-    },
-    count: {
-      type: Number,
-      default: 1,
-    },
-    linkFunction: {
-      type: Function,
-      default() {
-        return null
-      },
-    },
+
+const emit = defineEmits(['switch-page'])
```

Comment:
Likewise here, if you switched to TS, you could have a proper typing for this emit:
```suggestion
const emit = defineEmits<{
  'switch-page': [page: number]
}>()
```